components/salesforce/common.js

```diff
@@ -0,0 +1,151 @@
+      await this.salesforce.deleteApexClass(webhookId);
+
+      const { metadata: metadataApiUrl } = this.db.get("apiUrls");
+      const remoteSiteName = this.db.get("remoteSiteName");
+      await this.salesforce.deleteRemoteSite(metadataApiUrl, remoteSiteName);
+    },
+  },
+  methods: {
+    _getRemoteSiteName() {
+      const eventSourceName = this.getEventSourceName();
+      return this.standardizeName(eventSourceName);
+    },
+    _getTriggerName() {
+      return this.standardizeName("trigger");
+    },
+    _getWebhookName() {
+      return this.standardizeName("webhook");
+    },
+    _isValidSource(event) {
+      const {
+        "x-webhook-token": webhookToken,
+      } = event.headers;
+      const secretToken = this.db.get("secretToken");
+      return webhookToken === secretToken;
+    },
+    standardizeName(rawName) {
+      const options = {
+        replacement: "_",
+        remove: /[()-]/g,
+        lower: true,
+      };
+      const id = uuidv4();
+      const enrichedRawName = `pd_${rawName}_${id}`;
+      return slugify(enrichedRawName, options).substring(0, 40);
+    },
+    processEvent(event) {
+      const { body } = event;
+      const meta = this.generateMeta(event);
+      this.$emit(body, meta);
+    },
+    generateMeta() {
+      throw new Error('generateMeta is not implemented');
+    },
+    getEventSourceName() {
+      throw new Error('getEventSourceName is not implemented');
+    },
+    getEventTypes() {
+      throw new Error('getEventTypes is not implemented');
+    },
+    getObjectType() {
+      throw new Error('getObjectType is not implemented');
+    },
+    getTriggerBody() {
+      throw new Error('getTriggerBody is not implemented');
+    },
+    getWebhookBody(webhookName, secretToken) {
+      return `
+        public abstract class ${webhookName} {
+
+          public static String jsonContent(final Map<String, Object> eventContent) {
+              final Map<String, Object> content = new Map<String, Object>(eventContent);
+              content.put('UserId', UserInfo.getUserId());
+              return JSON.serialize(content);
+          }
+
+          @future(callout=true)
+          public static void callout(final String url, final String content) {
+              final HttpRequest request = new HttpRequest();
+              request.setEndpoint(url);
+              request.setMethod('POST');
+              request.setHeader('Content-Type', 'application/json');
+              request.setHeader('X-Webhook-Token', '${secretToken}');
+              request.setBody(content);
+
+              final Http http = new Http();
+              http.send(request);
+          }
+
+        }
+      `;
+    },
+  },
+  async run(event) {
+    if (!this._isValidSource(event)) {
+      this.http.respond({
+        statusCode: 404,
+      });
+      return;
+    }
+
+    this.http.respond({
+      statusCode: 200,
+    });
+
+    return this.processEvent(event);
```

Comment:
Do you need to `return` this or can you just call it?

Reply:
From the component's point of view, I could just do this and it would be the same:
```js
await this.processEvent(event);
```

I'll change to that for clarity.